Rakefile

```diff
@@ -425,3 +425,288 @@ end
+            name:                       'text',
+            help_msg:                   "The text to #{action}",
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        when 'project'
+          # bolt project init [directory] [options]
+          @pwsh_command[:options] << {
+            name:                       'directory',
+            help_msg:                   'The directory',
+            mandatory:                  false,
+            type:                       'string',
+            switch:                     false,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        end
+
+        help_text[:flags].reject { |o| o =~ /verbose|debug|help/ }.map do |option|
+          ruby_param = parser.top.long[option]
+          pwsh_name = option.gsub("-", "")
+
+          next if pwsh_name == 'version'
+
+          pwsh_param = {
+            name:       pwsh_name,
+            type:       'string',
+            switch:     false,
+            mandatory:  false,
+            help_msg:   ruby_param.desc.map { |d| d.gsub("<", "&lt;") }.join("\n"),
+            ruby_short: ruby_param.short.first,
+            ruby_long:  ruby_param.long.first,
+            ruby_arg:   ruby_param.arg,
+            ruby_orig:  option
+          }
+
+          case ruby_param.class.to_s
+          when 'OptionParser::Switch::RequiredArgument'
+            # this isn't quite the same thing as a mandatory pwsh parameter
+            # pwsh_param[:mandatory] = true
+          when 'OptionParser::Switch::OptionalArgument'
+            pwsh_param[:mandatory] = false
+          when 'OptionParser::Switch::NoArgument'
+            pwsh_param[:mandatory] = false
+            pwsh_param[:switch] = true
+            pwsh_param[:type] = 'switch'
+          end
+
+          # Only one of --targets , --rerun , or --query can be used
+          # Only one of --configfile or --boltdir can be used
+          case pwsh_name
+          when 'user'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'password'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'privatekey'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'concurrency'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'compileconcurrency'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'connecttimeout'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'modulepath'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'transport'
+            pwsh_param[:validate_set] = %w[winrm ssh pcp]
```

Comment:
I think all the transports will work on Windows? https://github.com/puppetlabs/bolt/blob/master/lib/bolt/config/options.rb#L15 the keys of that hash are the complete set, if you want to compute it

Reply:
My bad, I didn't look there. Thanks for pointing this out!

Reply:
It now adds all transports